### PR TITLE
feat(logs): add filter by pod in logs url

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-865

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28102cd2-684e-4ecb-a4f7-2f37f653b027 | [my-helm-env](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/28102cd2-684e-4ecb-a4f7-2f37f653b027/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Take rest sometimes

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] my-env - feat(logs): add filter by pod in logs url - 2024-01-22T17:01:20Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/7f2317ac-93a4-4815-876e-4c7d07bf3918/services/general)

 Another comment will be posted when deployments are terminated

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [\[PR\] my-env - feat(logs): add filter by pod in logs url - 2024-01-22T17:01:20Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/7f2317ac-93a4-4815-876e-4c7d07bf3918/logs)

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] staging - feat(logs): add filter by pod in logs url - 2024-01-23T08:33:16Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/230e298c-d70a-414f-b4e8-08cf0d77496c/services/general)

 Another comment will be posted when deployments are terminated

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://1142-console-staging.zc531a994.rustrocks.cloud)

File: libs/pages/logs/environment/src/lib/feature/pod-logs-feature/pod-logs-feature.tsx [hunk 1]
```diff
@@ -21,6 +22,39 @@ const PodLogs = memo(_PodLogs)
 
 export function PodLogsFeature({ clusterId }: PodLogsFeatureProps) {
   const { organizationId = '', projectId = '', environmentId = '', serviceId = '' } = useParams()
+  const [filter, setFilter] = useState<TableFilterProps[]>([])
+  const [searchParams, setSearchParams] = useSearchParams()
+
+  const POD_NAME_KEY = 'pod_name'
+  useEffect(() => {
```
Comment (RemiBonnet) [inline libs/pages/logs/environment/src/lib/feature/pod-logs-feature/pod-logs-feature.tsx:29]: I think it missing one behavior, if `filter` is undefined we need to reset `searchParams`, you can see on the Loom:

https://www.loom.com/share/0d53cdfdc43a4e5c8df9f580937639e2

**Review by RemiBonnet — CHANGES_REQUESTED** [review]

File: libs/pages/logs/environment/src/lib/feature/pod-logs-feature/pod-logs-feature.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/pages/logs/environment/src/lib/feature/pod-logs-feature/pod-logs-feature.tsx:29]: Fixed https://github.com/Qovery/console/pull/1142/commits/9f785d7080eba6040873bbaf99c6e3548480bbed

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**